README.md

```diff
@@ -6,6 +6,8 @@
 
 This repository contains the Prometheus storage layer that is used in its 2.x releases.
 
+It requires golang 1.12 or higher.
```

Comment:
Will we remember to keep this up to date?

Reply:
Is there any other way to enforce or notify the users of the the library that it requires 1.12 or higher?

Reply:
just found this and will try it
https://medium.com/@theckman/version-constraints-and-go-c9309be15773

Reply:
You could add `go 1.12` to `go.mod` though it won't prevent anybody to try building the library with an older version of Go. From https://golang.org/doc/go1.12#modules

> The go directive in a go.mod file now indicates the version of the language used by the files within that module. It will be set to the current release (go 1.12) if no existing version is present. If the go directive for a module specifies a version newer than the toolchain in use, the go command will attempt to build the packages regardless, and will note the mismatch only if that build fails.

Reply:
Aaah that is nice, thanks Simon.  
The description in godoc is a bit confusing though.

I implemented the suggested method from the blog post which will enforce a build failure for versions older than 1.12. 

Another approach would be to print a warning log in an init() func if the golang version is older than 1.12.

Reply:
I think best to leave the current enforcement and once golang progresses few more versions will switch to the go modules way.